content/doc/book/scaling/architecting-for-manageability.adoc

```diff
@@ -163,20 +163,25 @@ control our _$JENKINS_HOME_ in a Github repository known as "demo-joc". The
 
 [source,bash]
 ----
-FROM adoptopenjdk/openjdk8:debianslim
-RUN echo "deb [arch=amd64] https://download.docker.com/linux/debian buster stable" \
-     >> /etc/apt/sources.list.d/docker.list && \
-    apt-get update && apt-get -y upgrade && \
+FROM eclipse-temurin:11-jdk-jammy
+RUN mkdir -p /etc/apt/keyrings && \
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | \
+        gpg --dearmor -o /etc/apt/keyrings/docker.gpg && \
+    echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/docker.gpg] \
+        https://download.docker.com/linux/ubuntu jammy stable" \
+        >> /etc/apt/sources.list.d/docker.list > /dev/null && \
+    apt-get update && \
+    apt-get -y upgrade && \
```

Comment:
```suggestion
FROM eclipse-temurin:11-jdk-jammy
SHELL ["/bin/bash", "-o", "pipefail", "-c"]
RUN mkdir -p /etc/apt/keyrings && \
    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | \
        gpg --dearmor -o /etc/apt/keyrings/docker.gpg && \
    echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/docker.gpg] \
        https://download.docker.com/linux/ubuntu jammy stable" \
        >> /etc/apt/sources.list.d/docker.list 2>/dev/null && \
    apt-get update && \
    apt-get -y upgrade && \
```

- As a "piped" shell command is added, there is a need for specifying the pipefail option to the (`bash` since it's based on jammy) shell. Ref. https://github.com/hadolint/hadolint/wiki/DL4006
- there is an error when echo-ing the docker list in the apt list. Both `>` and `>>` operators are competing for the same output. It looks like (I might have misunderstood the intent) that you only want to redirect stderr to the `/dev/null` device: in that case, the suggestion shows a fix (missing `2` identifying the stderr before the `>` redirection).

(side note: I initially intented to remove the `apt upgrade` but while searching for documentation - https://github.com/hadolint/hadolint/wiki/DL3005 - I learned that in practise, it can be a bad idea as per https://github.com/hadolint/hadolint/issues/562 and hadolint disabled this rule)